internal/sequencerapi/api.go

```diff
@@ -0,0 +1,92 @@
+package sequencerapi
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+var (
+	sendRawTxConditionalCostMeter       = metrics.NewRegisteredMeter("sequencer/sendRawTransactionConditional/cost", nil)
+	sendRawTxConditionalRequestsCounter = metrics.NewRegisteredCounter("sequencer/sendRawTransactionConditional/requests", nil)
+	sendRawTxConditionalAcceptedCounter = metrics.NewRegisteredCounter("sequencer/sendRawTransactionConditional/accepted", nil)
+)
+
+type sendRawTxCond struct {
+	b ethapi.Backend
+}
+
+func GetSendRawTxConditionalAPI(b ethapi.Backend) rpc.API {
+	return rpc.API{
+		Namespace: "eth",
+		Service:   &sendRawTxCond{b},
+	}
+}
+
+func (s *sendRawTxCond) SendRawTransactionConditional(ctx context.Context, txBytes hexutil.Bytes, cond types.TransactionConditional) (common.Hash, error) {
+	sendRawTxConditionalRequestsCounter.Inc(1)
+
+	cost := cond.Cost()
+	sendRawTxConditionalCostMeter.Mark(int64(cost))
+	if cost > params.TransactionConditionalMaxCost {
+		return common.Hash{}, &rpc.JsonError{
+			Message: fmt.Sprintf("conditional cost, %d, exceeded max: %d", cost, params.TransactionConditionalMaxCost),
+			Code:    params.TransactionConditionalCostExceededMaxErrCode,
+		}
+	}
+
+	// Perform sanity validation prior to state lookups
+	if err := cond.Validate(); err != nil {
+		return common.Hash{}, &rpc.JsonError{
+			Message: fmt.Sprintf("failed conditional validation: %s", err),
+			Code:    params.TransactionConditionalRejectedErrCode,
+		}
+	}
+
+	header, err := s.b.HeaderByNumber(context.Background(), rpc.LatestBlockNumber)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	if err := header.CheckTransactionConditional(&cond); err != nil {
+		return common.Hash{}, &rpc.JsonError{
+			Message: fmt.Sprintf("failed header check: %s", err),
+			Code:    params.TransactionConditionalRejectedErrCode,
+		}
+	}
+
+	// State is checked against an older block to remove the MEV incentive for this endpoint compared with sendRawTransaction
+	parentBlock := rpc.BlockNumberOrHash{BlockHash: &header.ParentHash}
+	parentState, _, err := s.b.StateAndHeaderByNumberOrHash(context.Background(), parentBlock)
```

Comment:
Not worth checking against past state at all? How else might we discourage this endpoint against MEV vs sendRawTransaction?

Reply:
@protolambda what do you mean exactly? Are you saying the state check should only happen in the sequencer? You are right that there are consistency issues here, its a tradeoff, I think its better to do the check here, because really only high contention environments will break, which is ok because the searchers shouldn't use this endpoint

Reply:
I meant the usage of a parent block, instead of the already known latest block, here. The MEV question is interesting. Using older state might backfire more than we desire. How do other protocols handle this MEV edge? Is there a standard?

I'm fine with merging this as-is, but this parent-block choice is worth some discussion async.